scripts/generate-charts

```diff
@@ -31,6 +31,14 @@ for f in packages/*; do
 			cp -R ${f}/charts-crd/* charts/$(basename -- ${f})/charts-crd
 		fi
 		helm repo index --merge ./assets/index.yaml --url assets assets
+		
+		# Clean up changes
+		if [[ -f ${f}/package.yaml ]]; then
+			yq w -i ${f}/charts/Chart.yaml 'version' "${version}"
+		fi
+		for file in $(find ./${f}/overlay -type f); do
+			rm ${f}/charts/${file#./${f}/overlay/}
+		done
```

Comment:
This is already cleaned up with make clean isn't it ? Are these just left behind on patch ?

Reply:
This is the inverse of https://github.com/rancher/charts/blob/a5baa341654df0e333979f850c2277a95fb8b469/scripts/generate-charts#L7-L9

While it is cleaned up on `make clean`, [the original issue](https://github.com/rancher/rancher/issues/29064) was about making sure that a user who runs `make charts` followed by `make patch` doesn't end up leaving behind these overlaid resources (e.g. `make charts` should leave the contents of `packages/{chart}/*` identical to what it was before).

Reply:
It's a pretty common use case with the following developer workflow:
- Make changes to the chart and run `make patch`
- Run `make charts` (which also runs `make prepare`) and push your changes to your branch
- Add your branch to Rancher Dashboard and try installing your chart. See more changes that need to be made

At this step, currently you would have to run `make prepare` once more before trying to add the new changes and running `make patch`. With this fix, you could omit the extra `make prepare` step.

Reply:
>  With this fix, you could omit the extra make prepare step.

I don't understand that part, prepare doesn't do anything with overlays ?

Reply:
Yeah, it’s one of the things I wanted to fix during the rewrite. Adding support to using overlays on prepare would also require changes to the patch logic, but that’s another option we could take here.

Reply:
Ok, I guess this is fine, but we should limit changes to bash stuff.